src/components/nav/nav.tsx

```diff
@@ -0,0 +1,116 @@
+import { Component, Element, Event, EventEmitter, h, Host, Prop, VNode } from "@stencil/core";
+import { CSS } from "./resources";
+
+@Component({
+  tag: "calcite-nav",
+  styleUrl: "nav.scss",
+  shadow: true
+})
+
+/**
+ * @slot logo - A slot for adding a `calcite-logo` component to the primary nav level.
+ * @slot user - A slot for adding a `calcite-user` component to the primary nav level.
+ * @slot progress - A slot for adding a `calcite-progress` component to the primary nav level.
+ * @slot primary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the primary nav level.
+ * @slot primary-content-center - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the center position of the primary nav level.
+ * @slot primary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the primary nav level.
+ * @slot secondary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the secondary nav level.
+ * @slot secondary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the secondary nav level.
+ * @slot tertiary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the tertiary nav level.
+ * @slot tertiary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the tertiary nav level.
+ */
+export class CalciteNav {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteNavElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, display a hamburger  icon and emits `calciteNavMenuActionSelect` event on user interaction.
+   */
+  @Prop({ reflect: true }) displayMenuAction = false;
```

Comment:
open for suggestions regarding the prop name

Reply:
This actually renders a calcite-action, not an icon. Should the doc be updated?

Instead of having a prop for this, why not just let the user slot a calcite-action here? If we do need a name, how about just `menu` or `menuEnabled`?

Reply:
Have gone back and forth on slot vs. prop and supplied UI here - the usual pro/cons apply…

Providing this would lead to more consistent UX across platform, but of course necessitates accepting a label string, etc.

Reply:
Yeah supporting both a built-in “hamburger” and allowing folks to slot something else if they really need to makes sense to me.

Hesitant to call the prop only “menu” since we would only be rendering the “action that emits an event devs can use to open a Sheet / Panel” - “menu-toggle”?

Reply:
`menu-action` or `menu-toggle` makes sense to me

Reply:
I don't think theres any need to add `enable` onto the name. When you add it as a HTML attribute or set it to `true` that is already enabling it.

Reply:
`menu-action` sounds good - this will become more valuable once we have a Sheet component - have some ideas about leveraging that and a slot so folks won't have to set that up themselves but... this will be useful either way in the interim.

Reply:
Only other thought here is to update with `nav` terminology - `nav-action` - since it may or may not invoke a Menu.